crates/polars-core/src/chunked_array/cast.rs

```diff
@@ -174,12 +174,22 @@ impl ChunkCast for Utf8Chunked {
     fn cast(&self, data_type: &DataType) -> PolarsResult<Series> {
         match data_type {
             #[cfg(feature = "dtype-categorical")]
-            DataType::Categorical(_) => {
-                let iter = self.into_iter();
-                let mut builder = CategoricalChunkedBuilder::new(self.name(), self.len());
-                builder.drain_iter(iter);
-                let ca = builder.finish();
-                Ok(ca.into_series())
+            DataType::Categorical(rev_map) => match rev_map {
+                None => {
+                    let iter = self.into_iter();
```

Comment:
I know this was already here, but we could use the typed iterator here:

```rust
           let iter = unsafe {
                        self.downcast_iter()
                            .flatten()
                            .map(|x| x.copied())
                            .trust_my_length(self.len())
                    };
```

In a later PR we should change the signature for for `drain_iter` or add a `drain_iters` (`drain_iter` can be a subset of that).